### PR TITLE
Move systems and plugins to own modules 

### PR DESCRIPTION
**Pull request by Bjohnson131**
Hello, this is a simple change. 
It moves all plugins to a module called `plugins`, as well as all systems to a module called `systems`. I intend for this to be a bit more presentable than the flat hierarchy that we see now.

### CODE REVIEW

File: src/melee_attacker_system.rs [hunk 1]
```diff
@@ -1,11 +0,0 @@
-use crate::prelude::*;
-
-
-pub fn melee_attacks(
-    mut commands: Commands,
-    entities: Query<(Entity, &Position, With<MeleeAttacker>)>,
-) {
-    for (entity, position, _) in entities.iter() {
-
-    }
```
Comment (Bjohnson131) [inline src/melee_attacker_system.rs:10]: removed unused module.

File: src/music_system.rs [hunk 2]
```diff
@@ -1,20 +0,0 @@
-
-
-pub fn play_music(
-    app_state: Res<GameState>,
-) {
```
Comment (Bjohnson131) [inline src/music_system.rs:5]: removed unused module.

File: src/components.rs [hunk 3]
```diff
@@ -1,5 +1,5 @@
 use bevy::prelude::*;
-use super::prelude::*;
+use crate::prelude::*;
```
Comment (Bjohnson131) [inline src/components.rs:2]: super::prelude changed to crate::prelude it's more universal, and means the same thing throughout the project.

File: src/main.rs [hunk 4]
```diff
@@ -1,38 +1,41 @@
...
-    window_system
-)]
+mod plugins;
+mod systems;
+
+mod prelude;
+mod components;
+mod constants;
+mod load;
+mod resources;
```
Comment (Bjohnson131) [inline src/main.rs]: mod list is now much shorter.

**Review by ryankopf — COMMENTED** [review]
Good organization plan.

I think we should have left stubs for future intended features, like melee combat, but I'm not going to hold up this PR just for that.

**Comment by ryankopf** [comment]
Do you mind resolving the conflicts from the last PR and committing again?

Also, please double check that it builts before submitting a PR.
Shouldn't "crate::plugin::task" be "crate::plugins::task"? It's spelled wrong (missing the s) in several tasks.

**Comment by Bjohnson131** [comment]
Alright, I've fixed the conflicts.

**Comment by Bjohnson131** [comment]
Hey @ryankopf Instead of making these PRs, I'm going to file a few issues, and you can prioritize and select which ones are needed and which won't be done. Sound good?

**Comment by ryankopf** [comment]
Closing so we can work on issues one at a time

**Pull request closed without merging**